SundialsML 6.7.0 fails to compile: SUNADAPTCONTROLLER_* symbols not declared

# Summary:

The build fails at the make install step with errors related to undefined identifiers such as `SUNADAPTCONTROLLER_SUCCESS`, even though the build environment includes recent versions of Sundials and OCaml.

# Reproductibility

- **Sundials**: 6.7.0, 5.8.0

- **OCaml**: 5.0.0, 5.2.0, 5.4.0~alpha

- **OS**: macOS (M2 architecture) / Linux (Arch,  x86_64)

- **Build step**: make install


# Log output


```sh
make install
[...]
cc -I /Users/veco/.opam/sundialsml-5/lib/ocaml -isystem /usr/local/include/ -O3 -Wall -Werror -Wno-error=deprecated-declarations -DNDEBUG=1 -isystem /Users/veco/.opam/sundialsml-5/lib/ocaml -isystem /usr/local/include/  -o sundials/sundials_ml.o -c sundials/sundials_ml.c
sundials/sundials_ml.c:951:10: error: use of undeclared identifier 'SUNADAPTCONTROLLER_SUCCESS'
  951 |     case SUNADAPTCONTROLLER_SUCCESS:
      |          ^
sundials/sundials_ml.c:954:10: error: use of undeclared identifier 'SUNADAPTCONTROLLER_ILL_INPUT'
  954 |     case SUNADAPTCONTROLLER_ILL_INPUT:
      |          ^
sundials/sundials_ml.c:957:10: error: use of undeclared identifier 'SUNADAPTCONTROLLER_MEM_FAIL'
  957 |     case SUNADAPTCONTROLLER_MEM_FAIL:
      |          ^
sundials/sundials_ml.c:960:10: error: use of undeclared identifier 'SUNADAPTCONTROLLER_USER_FCN_FAIL'
  960 |     case SUNADAPTCONTROLLER_USER_FCN_FAIL:
      |          ^
sundials/sundials_ml.c:963:10: error: use of undeclared identifier 'SUNADAPTCONTROLLER_OPERATION_FAIL'
  963 |     case SUNADAPTCONTROLLER_OPERATION_FAIL:
      |          ^
sundials/sundials_ml.c:1372:5: error: non-void function 'sunml_adapt_reset' should return a value [-Wreturn-type]
 1372 |     CAMLreturn0;
      |     ^
/Users/veco/.opam/sundialsml-5/lib/ocaml/caml/memory.h:419:3: note: expanded from macro 'CAMLreturn0'
  419 |   return; \
      |   ^
sundials/sundials_ml.c:1382:5: error: non-void function 'sunml_adapt_set_defaults' should return a value [-Wreturn-type]
 1382 |     CAMLreturn0;
      |     ^
/Users/veco/.opam/sundialsml-5/lib/ocaml/caml/memory.h:419:3: note: expanded from macro 'CAMLreturn0'
  419 |   return; \
      |   ^
sundials/sundials_ml.c:1392:5: error: non-void function 'sunml_adapt_write' should return a value [-Wreturn-type]
 1392 |     CAMLreturn0;
      |     ^
/Users/veco/.opam/sundialsml-5/lib/ocaml/caml/memory.h:419:3: note: expanded from macro 'CAMLreturn0'
  419 |   return; \
      |   ^
sundials/sundials_ml.c:1403:5: error: non-void function 'sunml_adapt_set_error_bias' should return a value [-Wreturn-type]
 1403 |     CAMLreturn0;
      |     ^
/Users/veco/.opam/sundialsml-5/lib/ocaml/caml/memory.h:419:3: note: expanded from macro 'CAMLreturn0'
  419 |   return; \
      |   ^
sundials/sundials_ml.c:1414:5: error: non-void function 'sunml_adapt_update_h' should return a value [-Wreturn-type]
 1414 |     CAMLreturn0;
      |     ^
/Users/veco/.opam/sundialsml-5/lib/ocaml/caml/memory.h:419:3: note: expanded from macro 'CAMLreturn0'
  419 |   return; \
      |   ^
10 errors generated.
make[1]: *** [sundials/sundials_ml.o] Error 1
make: *** [install] Error 2
```

--- a/src/sundials/sundials_ml.c
+++ b/src/sundials/sundials_ml.c
@@ -945,6 +945,7 @@ CAMLprim value sunml_context_create_parallel(value vcomm)
 
 /* Adaptivity Controllers */
 
+#if 670 <= SUNDIALS_LIB_VERSION
 void sunml_adapt_check_flag( const char* call, int flag)
 {
     switch (flag) {
@@ -967,6 +968,7 @@ void sunml_adapt_check_flag( const char* call, int flag)
 	caml_failwith("sunml_adapt_check_flag");
     }
 }
+#endif
 
 #if 670 <= SUNDIALS_LIB_VERSION
 static void finalize_adapt(value vadaptc)
@@ -1369,7 +1371,7 @@ CAMLprim value sunml_adapt_reset(value vadaptc)
     int r = SUNAdaptController_Reset(ML_ADAPTCONTROLLER(vadaptc));
     CHECK_ADAPT_FLAG("SUNAdaptController_Reset", r);
 #endif
-    CAMLreturn0;
+    CAMLreturn(Val_unit);
 }
 
 CAMLprim value sunml_adapt_set_defaults(value vadaptc)
@@ -1379,7 +1381,7 @@ CAMLprim value sunml_adapt_set_defaults(value vadaptc)
     int r = SUNAdaptController_SetDefaults(ML_ADAPTCONTROLLER(vadaptc));
     CHECK_ADAPT_FLAG("SUNAdaptController_SetDefaults", r);
 #endif
-    CAMLreturn0;
+    CAMLreturn(Val_unit);
 }
 
 CAMLprim value sunml_adapt_write(value vadaptc, value vlog)
@@ -1389,7 +1391,7 @@ CAMLprim value sunml_adapt_write(value vadaptc, value vlog)
     int r = SUNAdaptController_Write(ML_ADAPTCONTROLLER(vadaptc), ML_CFILE(vlog));
     CHECK_ADAPT_FLAG("SUNAdaptController_Write", r);
 #endif
-    CAMLreturn0;
+    CAMLreturn(Val_unit);
 }
 
 CAMLprim value sunml_adapt_set_error_bias(value vadaptc, value vbias)
@@ -1400,7 +1402,7 @@ CAMLprim value sunml_adapt_set_error_bias(value vadaptc, value vbias)
 					    Double_val(vbias));
     CHECK_ADAPT_FLAG("SUNAdaptController_SetErrorBias", r);
 #endif
-    CAMLreturn0;
+    CAMLreturn(Val_unit);
 }
 
 CAMLprim value sunml_adapt_update_h(value vadaptc, value vh, value vdsm)
@@ -1411,7 +1413,7 @@ CAMLprim value sunml_adapt_update_h(value vadaptc, value vh, value vdsm)
 				       Double_val(vh), Double_val(vdsm));
     CHECK_ADAPT_FLAG("SUNAdaptController_UpdateH", r);
 #endif
-    CAMLreturn0;
+    CAMLreturn(Val_unit);
 }
 
 CAMLprim value sunml_adapt_space(value vadaptc)

